integration-tests/src/test/java/com/datastax/oss/driver/api/core/metadata/NodeStateIT.java

```diff
@@ -0,0 +1,381 @@
+
+      localSimulacronNode.rejectConnections(0, RejectScope.UNBIND);
+      localSimulacronNode.closeConnections(CloseType.DISCONNECT);
+
+      ConditionChecker.checkThat(
+              () ->
+                  localMetadataNode.getState() == NodeState.DOWN
+                      && localMetadataNode.isReconnecting())
+          .as("Node going down")
+          .before(10, TimeUnit.SECONDS)
+          .becomesTrue();
+
+      expect(NodeStateEvent.changed(NodeState.UP, NodeState.DOWN, localMetadataNode));
+
+      localSimulacronNode.acceptConnections();
+      ((InternalDriverContext) localCluster.getContext())
+          .eventBus()
+          .fire(TopologyEvent.suggestUp(localMetadataNode.getConnectAddress()));
+
+      ConditionChecker.checkThat(
+              () ->
+                  localMetadataNode.getState() == NodeState.UP
+                      && !localMetadataNode.isReconnecting())
+          .as("Node coming back up")
+          .before(10, TimeUnit.SECONDS)
+          .becomesTrue();
+
+      expect(NodeStateEvent.changed(NodeState.DOWN, NodeState.UP, localMetadataNode));
+    }
+  }
+
+  @Test
+  public void should_force_down_when_not_ignored() throws InterruptedException {
+    driverContext.eventBus().fire(TopologyEvent.forceDown(metadataRegularNode.getConnectAddress()));
+    ConditionChecker.checkThat(
+            () ->
+                metadataRegularNode.getState() == NodeState.FORCED_DOWN
+                    && metadataRegularNode.getOpenConnections() == 0
+                    && !metadataRegularNode.isReconnecting())
+        .as("Node forced down")
+        .before(10, TimeUnit.SECONDS)
+        .becomesTrue();
+
+    // Should ignore up/down topology events while forced down
+    driverContext.eventBus().fire(TopologyEvent.suggestUp(metadataRegularNode.getConnectAddress()));
+    TimeUnit.MILLISECONDS.sleep(200);
+    assertThat(metadataRegularNode).hasState(NodeState.FORCED_DOWN);
+
+    driverContext
+        .eventBus()
+        .fire(TopologyEvent.suggestDown(metadataRegularNode.getConnectAddress()));
+    TimeUnit.MILLISECONDS.sleep(200);
+    assertThat(metadataRegularNode).hasState(NodeState.FORCED_DOWN);
+
+    // Should only come back up on a FORCE_UP event
+    driverContext.eventBus().fire(TopologyEvent.forceUp(metadataRegularNode.getConnectAddress()));
+    ConditionChecker.checkThat(
+            () ->
+                metadataRegularNode.getState() == NodeState.UP
+                    && metadataRegularNode.getOpenConnections() == 2
+                    && !metadataRegularNode.isReconnecting())
+        .as("Node forced back up")
+        .before(10, TimeUnit.SECONDS)
+        .becomesTrue();
+  }
+
+  @Test
+  public void should_force_down_when_ignored() throws InterruptedException {
+    driverContext
+        .loadBalancingPolicyWrapper()
+        .setDistance(metadataRegularNode, NodeDistance.IGNORED);
```

Comment:
👍 This is much cleaner, and actually faster than waiting for the cluster to get back to its base state each time.